python/TestHarness/testers/Tester.py

```diff
@@ -41,6 +41,8 @@ def validParams():
     params.addParam('petsc_debug',   ['ALL'], "{False,True} -> test only runs when PETSc is configured with --with-debugging={0,1}, otherwise test always runs.")
     params.addParam('curl',          ['ALL'], "A test that runs only if CURL is detected ('ALL', 'TRUE', 'FALSE')")
     params.addParam('tbb',           ['ALL'], "A test that runs only if TBB is available ('ALL', 'TRUE', 'FALSE')")
+    params.addParam('pthreads',      ['ALL'], "A test that runs only if PTHREADS is available ('ALL', 'TRUE', 'FALSE')")
+    params.addParam('openmp',        ['ALL'], "A test that runs only if OPENMP is available ('ALL', 'TRUE', 'FALSE')")
```

Comment:
This is misleading. Right now libMesh is often configured with multiple thread models enabled and they have a set precedence. As a user though, I might want to make a test run with a specific model like "pthreads". I might think I want to turn on "pthreads = true". However, this test will run when I'm using TBB, OpenMP as long as pthreads is also turned on. I know why you are doing this but I don't think the interface is quite right. I'm open to ideas on how to make this more intuitive.